packages/eslint-plugin/README.md

```diff
@@ -73,25 +88,76 @@ function myRuleFunction(context) {
 }
 ```
 
+> This example explicitly types the `context` and `literalNode` parameters, but these can be automatically inferred by Typescript
+
+In addition to callbacks that trigger on a simple node type (`Literal` in the example above),
+you can also trigger a callback using an [eslint selector](https://eslint.org/docs/developer-guide/selectors).
+
+Think of an eslint selector as a CSS selector, but for an AST.
+Selectors can reduce boilerplate while writing a rule, but more importantly they can potentially improve readability.
+
+```ts
+// Using a selector
+function myRuleFunction(context: Rule.RuleContext) {
+  return {
+    // Find an ExpressionStatement
+    // - that is a CallExpression
+    //   - that has a callee object named 'React'
+    //   - and has a callee property named 'useEffect'
+    "ExpressionStatement > CallExpression[callee.object.name = 'React'][callee.property.name = 'useEffect']"(
+      node: ExpressionStatement,
+    ) {
+      const message = 'Prefer bare useEffect() over React.useEffect()';
+      context.report({ node, message });
+    },
+  };
+}
+
+// Not using a selector
+function myRuleFunction(context: Rule.RuleContext) {
+  return {
+    ExpressionStatement(node) {
+      const expression = node.expression;
+      if (
+        expression?.type === 'CallExpression' &&
+        expression.callee.type === 'MemberExpression' &&
+        expression.callee.object.name === 'React' &&
+        expression.callee.property.name === 'useEffect'
+      ) {
+        const message = 'Prefer bare useEffect() over React.useEffect()';
```

Comment:
Why?

Reply:
This isn't intending to be an actual rule, but only demonstrate HOW you might write a rule.  I chose this useEffect example because I thought it would be simple to read and understand the implementation.